dashboard/static/js/controllers.js

```diff
@@ -1,14 +1,24 @@
 visApp.controller('mainController', ['$scope', 'Get','$location', '$anchorScroll', function($scope, Get,$location,$anchorScroll) {
 
-   function dataCall(){
-        var dataFile = '/api/v1/courses/'+dashboard.course_id+'/assignment_view?percent='+$scope.percent_filter;
+   function dataCall(call){
+        var dataFile = '/api/v1/courses/'+dashboard.course_id+'/assignments?percent='+$scope.percent_filter;
         Get.getData(dataFile).then(function(data) {
-            $scope.data = data.data
-            scrollId=1
-            $scope.data.find(function (e) {
-                current_week_indicator=e.due_date_items[0].assignment_items[0].current_week
-                if(current_week_indicator){
-                    return scrollId=e.id;
+            if (_.isEmpty(data.data)) {
+                $scope.plan = []
+                $scope.progress = []
+                return;
+            }
+            $scope.plan = data.data.plan
+            $scope.progress = data.data.progress
+            // call the js file onload only
+            if (call == 1) {
```

Comment:
This would probably be clearer if you used an object lookup. Maybe like
```

var dataScripts = {
    progress: '/static/js/assign-progress.js',
}
```

Then you could call it as dataCall("progress") instead of dataCall(1) and look it up in this object? The way it is now it's not clear what calling this with anything does. I think it would even be better if you just called it with the name of the script like dataCall ("assign-progress.js") rather than an integer. Then it can concat to /static/js ?

Reply:
Ok, I changed it as per the second suggestion